### PR TITLE
Include_adapter

### PR DESCRIPTION
**Pull request by Basliel-Amsalu**
implemented an adapter for transcript includes exon relationship based on the genecode data

### CODE REVIEW

File: config/adapters_config.yaml [hunk 1]
```diff
@@ -424,6 +424,17 @@ gencode_exon:
   nodes: True
   edges: False
 
+includes:
```
Comment (dawit-melka) [inline config/adapters_config.yaml:427]: You should use a more descriptive name like `transcript_includes_exon`

File: config/adapters_config_sample.yaml [hunk 2]
```diff
@@ -365,6 +365,17 @@ gencode_exon:
   nodes: True
   edges: False
 
+includes:
```
Comment (dawit-melka) [inline config/adapters_config_sample.yaml:368]: Same here

File: biocypher_metta/adapters/gencode_exon_adapter.py [hunk 3]
```diff
@@ -76,3 +76,34 @@ def get_nodes(self):
...
+                if data_line[GencodeExonAdapter.INDEX['type']] != 'exon':
+                    continue
+
+                info = self.parse_info_metadata(data_line[GencodeExonAdapter.INDEX['info']:])
+                transcript_key = info['transcript_id'].split('.')[0]
+                if info['transcript_id'].endswith('_PAR_Y'):
+                    transcript_key = transcript_key + '_PAR_Y'
+                exon_key = info['exon_id'].split('.')[0]
+                if info['exon_id'].endswith('_PAR_Y'):
+                    exon_key = exon_key + '_PAR_Y'
```
Comment (dawit-melka) [inline biocypher_metta/adapters/gencode_exon_adapter.py:95]: Can you also add this part in the `get_nodes` of exon?

File: biocypher_metta/adapters/gencode_exon_adapter.py [hunk 4]
```diff
@@ -76,3 +76,34 @@ def get_nodes(self):
...
+
+                _props = {}
+                if self.write_properties and self.add_provenance:
+                    _props['source'] = self.source
+                    _props['source_url'] = self.source_url
+
+                try:
+                    _id = transcript_key + '_' + exon_key
```
Comment (dawit-melka) [inline biocypher_metta/adapters/gencode_exon_adapter.py:103]: what is this `_id` used for?

File: config/schema_config.yaml [hunk 5]
```diff
@@ -886,4 +886,12 @@ histone modification:
   source: snp
   target: uberon #TODO replace by cell ontology
   properties:
-    modification: str
\ No newline at end of file
+    modification: str
+  
+transcript_to_exon:
+  is_a: related to at instance
+  description:
```
Comment (dawit-melka) [inline config/schema_config.yaml:893]: please add a short description about this relation

File: biocypher_metta/adapters/gencode_exon_adapter.py [hunk 6]
```diff
@@ -18,13 +18,13 @@ class GencodeExonAdapter(Adapter):
...
 
-    def __init__(self, write_properties, add_provenance, filepath=None,
+    def __init__(self, write_properties, label, add_provenance, filepath=None,
                  chr=None, start=None, end=None):
         self.filepath = filepath
         self.chr = chr
         self.start = start
         self.end = end
-        self.label = 'exon'
+        self.label = label if label else 'exon'
```
Comment (dawit-melka) [inline biocypher_metta/adapters/gencode_exon_adapter.py:27]: instead you can put `label='exon'` in the constructor argument. But you should switch the order with `add_provenance`.
`def __init__(....,add_provenance, label='exon', ...)`

**Review by dawit-melka — CHANGES_REQUESTED** [review]

File: biocypher_metta/adapters/gencode_exon_adapter.py (hunk 4 above)
Comment (Basliel-Amsalu) [reply biocypher_metta/adapters/gencode_exon_adapter.py:103]: it's a  mistake

File: biocypher_metta/adapters/gencode_exon_adapter.py (hunk 3 above)
Comment (Basliel-Amsalu) [reply biocypher_metta/adapters/gencode_exon_adapter.py:95]: i wasn't sure if there are any exons that end with PAR_Y and it is not on the sample but i added it just to be safe

**Review by dawit-melka — CHANGES_REQUESTED** [review]

File: config/schema_config.yaml [hunk 7]
```diff
@@ -894,3 +894,13 @@ histone modification:
...
+    
+transcript_to_exon:
+  is_a: related to at instance
+  description: >-
+    An association between a transcript and its exons
+  represented_as: edge
+  input_label: includes
+  source: transcript
+  target: exon
+
```
Comment (dawit-melka) [inline config/schema_config.yaml:906]: Please add source and source_url as a property

**Review by dawit-melka — CHANGES_REQUESTED** [review]

**Pull request merged**